### DISCUSSION TITLE
delegate security policy

### DISCUSSION THREAD

**Discussion by pgajdos** [ImageMagick/ImageMagick#7145] [General]
I have
```
$ convert terminal-distrobox-icon.svg -resize 16x16 terminal-distrobox-icon.png
convert: attempt to perform an operation not allowed by the security policy `inkscape' @ error/delegate.c/ExternalDelegateCommand/387.
$
```

So I did fx.
```
  <policy domain="delegate" rights="none" pattern="*"/>
  <policy domain="delegate" rights="read|write" pattern="inkscape"/>
```
or just
```
<policy domain="delegate" rights="read|write" pattern="*"/>
```

but that didn't helped. Only removing the delegate policy line works. What is the current recommendation?

**Comment by fmw42** [comment]
You should have for svg:decode and using Inkscape if you installed inscape properly.

  <delegate decode="svg:decode" stealth="True" command="&quot;inkscape&quot; &quot;%s&quot; --export-filename=&quot;%s&quot; --export-dpi=&quot;%s&quot; --export-background=&quot;%s&quot; --export-background-opacity=&quot;%s&quot; &gt; &quot;%s&quot; 2&gt;&amp;1"/>

If that does not work, then modify it to put the full path to Inkscape

**Comment by pgajdos** [comment]
Hi ho Fred (@fmw42),

thank you for your reply. My intention for now is to relax
```
<policy domain="delegate" rights="none" pattern="*"/>
```
to allow inkscape delegate. Makes this sense? Is that possible and how, exactly?

Yes, when I use just
```
<policy domain="delegate" rights="read|write" pattern="/usr/bin/inkscape"/>
```
it is better and I can now decode svg images, but I assume I do not restrict other delegates. When I use

```
  <policy domain="delegate" rights="none" pattern="*"/>
  <policy domain="delegate" rights="read|write" pattern="/usr/bin/inkscape"/>
```
the command in question fails with the policy error again. Please pardon my ignorance and elaborate on svg:decode, didn't get it.

**Comment by fmw42** [comment]
You misunderstand 

`<policy domain="delegate" rights="none" pattern="*"/>`

That prevents users from executing any image filters.

And

`<policy domain="path" rights="read" pattern="/repository/*" />`

I believe means that the glob expression is used to match all paths that start with /repository, where you have to fill in some delegate name as "repository"

**Comment by fmw42** [comment]
`You should have for svg:decode and using Inkscape if you installed inscape properly.
`

When you run your magick image.svg ... etc
put in -verbose after magick, so  magick -verbose image.svg ... etc

What do you get back?  That should show what delegate is being used.

Also if you look in delegates.xml file, you should find something like

<delegate decode="svg:decode" stealth="True" command="&quot;inkscape&quot; &quot;%s&quot; --export-filename=&quot;%s&quot; --export-dpi=&quot;%s&quot; --export-background=&quot;%s&quot; --export-background-opacity=&quot;%s&quot; &gt; &quot;%s&quot; 2&gt;&amp;1"/>

which shows Inkscape as being used for the command in the svg:decode line of the file.

**Comment by pgajdos** [comment]
Yep, I feel I didn't get something around. I just wanted to be close to policy-secure.xml as possible, so I wanted to keep 
```
<policy domain="delegate" rights="none" pattern="*"/>
```
but allow exceptions, e. g. inkscape and others reasonable which would pop up along. If it is not possible, I will just remove this line and be done for now, as I do not have real requirement for this attitude yet. I will perhaps return to it later.

**Comment by laighside** [comment]
For anyone else looking for the answer to block all delegates except inkscape, this seems to work:
```
<policy domain="delegate" rights="none" pattern="*"/>
<policy domain="delegate" rights="execute" pattern="inkscape"/>
```
The rights to enable delegates are `execute` not `read|write`

**Comment by pgajdos** [comment]
Thanks @laighside
